cdbot/cogs/maths.py

```diff
@@ -144,15 +144,26 @@ async def challenge(self, ctx: Context, number: int = 1):
         return await ctx.send(embed=embed)
 
     @command()
-    async def latex(self, ctx: Context, *, expression: str):
+    async def latex(self, ctx: Context):
         """
         Render a LaTeX expression with https://quicklatex.com/
         """
-        await self.latex_render(ctx, expression)
+        await self.latex_render(ctx, ctx.message)
 
-    async def latex_render(self, ctx: Context, expression: str):
+    async def latex_render(self, ctx: Context, message: Message):
+
+        # When invoked with the word latex, it is run as a command so ctx is type Context
+        # but when invoked with just $$$$ the context is a TextChannel
+        # so we need to check ctx's type
         channel = ctx.channel.id if type(ctx) is Context else ctx.id
 
+        # get the latex expression to be rendered
+        expression = message.content
+        if message.content.startswith("...latex "):
+            expression = message.content.replace("...latex ", "")
+        elif message.content.startswith(":latex "):
+            expression = message.content.replace(":latex ", "")
```

Comment:
```suggestion
expression = message.content

if type(ctx) is Context:
    channel = ctx.channel.id
    for prefix in "...latex ", ":latex ":
        expression = expression.replace(prefix, "")
else:
    channel = ctx.id
```

Reply:
Remember that the bot can also be invoked by tagging it at the start of a message.

Reply:
Would prefer it if there was some context method to strip prefixes but meh

Reply:
I am sure there's a cleaner way to handle this (possibly by not calling the same function for commands and non-commands?) but that can go in a different PR I'm sure
![image](https://user-images.githubusercontent.com/25161069/85571101-cc8f1700-b62b-11ea-9ea6-816e7fd781dd.png)